src/storage/conversion/GenericRdfToQuadConverter.ts

```diff
@@ -0,0 +1,50 @@
+import { PassThrough, Readable } from 'stream';
+import rdfParser from 'rdf-parse';
+import { Representation } from '../../ldp/representation/Representation';
+import { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import { CONTENT_TYPE_QUADS, DATA_TYPE_QUAD } from '../../util/ContentTypes';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+import { checkRequest } from './ConversionUtil';
+import { RepresentationConverter, RepresentationConverterArgs } from './RepresentationConverter';
+
+/**
+ * Converts most major RDF serializations to `internal/quads`.
+ */
+export class GenericRdfToQuadConverter extends RepresentationConverter {
+  public async canHandle(input: RepresentationConverterArgs): Promise<void> {
+    checkRequest(input, await rdfParser.getContentTypes(), [ CONTENT_TYPE_QUADS ]);
+  }
+
+  public async handle(input: RepresentationConverterArgs): Promise<Representation> {
+    return this.turtleToQuads(input.representation, input.identifier.path);
+  }
+
+  private turtleToQuads(representation: Representation, baseIRI: string): Representation {
+    const metadata: RepresentationMetadata = { ...representation.metadata, contentType: CONTENT_TYPE_QUADS };
+
+    // Catch parsing errors and emit correct error
+    // Node 10 requires both writableObjectMode and readableObjectMode
+    const errorStream = new PassThrough({ writableObjectMode: true, readableObjectMode: true });
+    let data: Readable;
+    try {
+      data = rdfParser.parse(representation.data, {
+        ...representation.metadata.contentType ?
+          { contentType: representation.metadata.contentType } :
+          { path: baseIRI },
```

Comment:
`checkRequest` in the `canHandle` call above is going to error if there is no contentType so you can assume it exists. The Solid spec currently says to reject PUT/POST/PATCH without content-type as well.

Reply:
Oh ok, that makes our lives easier here :)